### PR TITLE
Add flutter CI

### PR DESCRIPTION
**Pull request by epicfaace**
Testing CI for flutter

### CODE REVIEW

**Comment by dnfield** [comment]
This is failing because the test doesn't work even locally right now.  If you want it to just be a very basic smoke test, you can change `widget_test.dart` to read:

```dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter_test/flutter_test.dart';

import 'package:WHOFlutter/main.dart';

void main() {
  testWidgets('App Smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(MyApp());
  });
}
```

File: .github/workflows/flutter.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '12.x'
+      - uses: subosito/flutter-action@v1
```
Comment (SamMousa) [inline .github/workflows/flutter.yml]: We should not use 3rd party actions on the master, they have actions to secrets.

File: .github/workflows/flutter.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+jobs:
+  test:
+    name: Test on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+      - uses: actions/checkout@v1
```
Comment (SamMousa) [inline .github/workflows/flutter.yml]: Use `@v2`

File: .github/workflows/flutter.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '12.x'
+      - uses: subosito/flutter-action@v1
+        with:
+          channel: 'stable'
```
Comment (SamMousa) [inline .github/workflows/flutter.yml]:
```suggestion
      - echo "::set-env name=FLUTTER_REVISION::$(grep revision: .metadata | sed 's/  revision: //g')"
      - uses: actions/checkout@v2
        with:
          repository: 'flutter/flutter'
          ref: $FLUTTER_REVISION
          path: '_flutter'
      - run: echo "::add-path::$GITHUB_WORKSPACE/_flutter/bin"
```

File: .github/workflows/flutter.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '12.x'
+      - echo "::set-env name=FLUTTER_REVISION::$(grep revision: .metadata | sed 's/  revision: //g')"
```
Comment (SamMousa) [inline .github/workflows/flutter.yml]:
```suggestion
      - echo '::set-env name=FLUTTER_REVISION::$(grep revision: .metadata | sed "s/  revision: //g")'
```

**Comment by SamMousa** [comment]
Moving this to a branch on the main repo to make testing the CI a bit easier

**Pull request closed without merging**